src/main/java/org/apache/commons/lang3/math/NumberUtils.java

```diff
@@ -1822,4 +1824,125 @@ public static int compare(final short x, final short y) {
     public static int compare(final byte x, final byte y) {
         return x - y;
     }
+
+    /**
+     * <p>Checks if a {@code Number} value is zero,
+     * handling {@code null} by returning {@code false}.</p>
+     *
+     * @param number the {@code number} to check, {@code null} returns {@code false}
+     * @return {@code true} only if the input is non-null and equals zero
+     * @since 3.10
+     */
+    public static boolean isZero(final Number number) {
+        if (Objects.isNull(number)) {
+            return false;
+        } else if (number instanceof Integer) {
+            return number.intValue() == INTEGER_ZERO;
+        } else if (number instanceof Long) {
+            return number.longValue() == LONG_ZERO;
+        } else if (number instanceof Byte) {
+            return number.byteValue() == BYTE_ZERO;
+        } else if (number instanceof Short) {
+            return number.shortValue() == SHORT_ZERO;
+        } else if (number instanceof Float) {
+            return number.floatValue() == FLOAT_ZERO;
+        } else if (number instanceof Double) {
+            return number.doubleValue() == DOUBLE_ZERO;
+        }
+
+        return false;
```

Comment:
This is bogus IMO, if I create a subclass of `Number`, this method will always return false even if my `Number`'s inValue() returns 0. Why is this whole method's implementation not simply `return number.doubleValue() == DOUBLE_ZERO` ?

Reply:
agree, updated the func.